torch/csrc/distributed/autograd/context/dist_autograd_container.cpp

```diff
@@ -173,6 +180,22 @@ void DistAutogradContainer::eraseContextIdAndReset(int64_t context_id) {
   }
 }
 
+void DistAutogradContainer::cleanupContextWatchdog() {
+  std::this_thread::sleep_for(kContextTimeout);
+  while(!terminateWatchdog_.load()) {
+    std::lock_guard<std::mutex> guard(autograd_context_lock_);
+    for (auto& pair : autograd_context_) {
+      if (autograd_context_.find(pair.first) != autograd_context_.end()) { 
```

Comment:
Do we need this `if` check? We're iterating through `autograd_context_`, so this should always be true